docs/MAINTAINERS.md

```diff
@@ -1,52 +1,180 @@
 Maintaining Etcher
 ==================
 
-This document is meant to serve as a guide for maintainers to perform common
-tasks.
+This document is meant to serve as a guide for maintainers to perform common tasks.
 
-Preparing a new version
------------------------
+Releasing
+---------
 
-- Bump the version number in the `package.json`'s `version` property.
+### Release Types
 
-- Bump the version number in the `npm-shrinkwrap.json`'s `version` property.
+- **snapshot** (default): Describe me!
+- **production**: Describe me too!
 
-- Add a new entry to `CHANGELOG.md` by running `make changelog`.
+### Flight Plan
 
-- Update `screenshot.png` so it displays the latest version in the bottom
-right corner.
+#### Preparation
+
+- [Prepare the new version](#preparing-a-new-version)
+- [Generate build artefacts](#generating-binaries) (binaries, archives, etc.)
+- [Draft a release on GitHub](https://github.com/resin-io/etcher/releases)
+    - Upload build artefacts to GitHub release draft
+
+#### Testing
+
+- Test the prepared release and build artefacts properly on **all supported operating systems** to prevent regressions that went uncaught by the CI tests
+- If regressions or other issues arise, create issues on the repository for each one, and decide whether to fix them in this release (meaning repeating the process up until this point), or to follow up with a patch release
+
+#### Publishing
 
+- [Publish release draft on GitHub](https://github.com/resin-io/etcher/releases)
+- [Update the website](https://github.com/resin-io/etcher-homepage)
+- [Post release note to forums](https://forums.resin.io/c/etcher)
+- [Upload deb & rpm packages to Bintray](#uploading-packages-to-bintray)
+- [Upload build artefacts to Amazon S3](#uploading-binaries-to-amazon-s3)
+- Post changelog with `#release-notes` tag on Flowdock
+- If this release packs noteworthy major changes, maybe write a blog post about it as well
```

Comment:
Maybe worth mentioning the Etcher mailing list?

Reply:
It's just occurred to me that it might also be worth sending out a tweet announcing the new Etcher version on the https://twitter.com/resin_io Twitter account?

Reply:
Yup, added the mailing list